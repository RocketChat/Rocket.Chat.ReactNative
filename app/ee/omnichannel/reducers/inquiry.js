import { INQUIRY } from '../../../actions/actionsTypes';

const initialState = {
	enabled: false,
	queued: [],
	error: {}
};

export default function inquiry(state = initialState, action) {
	switch (action.type) {
		case INQUIRY.SUCCESS:
			return {
				...state,
				queued: action.inquiries
			};
		case INQUIRY.FAILURE:
			return {
				...state,
				error: action.error
			};
		case INQUIRY.SET_ENABLED:
			return {
				...state,
				enabled: action.enabled
			};
		case INQUIRY.QUEUE_ADD:
			return {
				...state,
				queued: [...state.queued, action.inquiry]
			};
		case INQUIRY.QUEUE_UPDATE:
			return {
				...state,
				queued: state.queued.map((item) => {
					if (item._id === action.inquiry._id) {
						return action.inquiry;
					}
					return item;
				})
			};
		case INQUIRY.QUEUE_REMOVE:
			return {
				...state,
				queued: state.queued.filter(({ _id }) => _id !== action.inquiryId)
			};
		case INQUIRY.RESET:
			return initialState;
		default:
			return state;
	}
}
