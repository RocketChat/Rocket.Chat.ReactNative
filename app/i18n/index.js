import i18n from 'i18n-js';
import { I18nManager } from 'react-native';
import * as RNLocalize from 'react-native-localize';

import en from './locales/en';
import ru from './locales/ru';
import fr from './locales/fr';
import de from './locales/de';
import nl from './locales/nl';
import ptBR from './locales/pt-BR';
import zhCN from './locales/zh-CN';
import ptPT from './locales/pt-PT';
import esES from './locales/es-ES';
import it from './locales/it';
import ja from './locales/ja';

i18n.translations = {
	en,
	ru,
	'pt-BR': ptBR,
	'zh-CN': zhCN,
	fr,
	de,
	'pt-PT': ptPT,
	'es-ES': esES,
	nl,
	it,
	ja
};
i18n.fallbacks = true;

const defaultLanguage = { languageTag: 'en', isRTL: false };
const availableLanguages = Object.keys(i18n.translations);
const { languageTag, isRTL } = RNLocalize.findBestAvailableLanguage(availableLanguages) || defaultLanguage;

I18nManager.forceRTL(isRTL);
i18n.locale = languageTag;

export default i18n;
