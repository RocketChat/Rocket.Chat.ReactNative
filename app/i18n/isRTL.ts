// https://github.com/zoontek/react-native-localize/blob/master/src/constants.ts#L5
const USES_RTL_LAYOUT = ['ar', 'ckb', 'fa', 'he', 'ks', 'lrc', 'mzn', 'ps', 'ug', 'ur', 'yi'];

export const isRTL = (locale?: string) => (locale ? USES_RTL_LAYOUT.includes(locale) : false);
