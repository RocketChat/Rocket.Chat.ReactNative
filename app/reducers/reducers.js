import * as types from '../constants/types';
import initialState from './initialState';

export default function settings(state = initialState.settings, action) {
	if (action.type === types.ADD_SETTINGS) {
		return {
			...state,
			...action.payload
		};
	}
	return state;
}
