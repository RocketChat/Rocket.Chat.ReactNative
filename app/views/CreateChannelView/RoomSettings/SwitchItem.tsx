import React from 'react';
import { StyleSheet, Switch, Text, View, SwitchProps } from 'react-native';

import I18n from '../../../i18n';
import { SWITCH_TRACK_COLOR } from '../../../lib/constants';
import { useTheme } from '../../../theme';
import sharedStyles from '../../Styles';

const styles = StyleSheet.create({
	switchContainer: {
		minHeight: 54,
		alignItems: 'center',
		justifyContent: 'space-between',
		flexDirection: 'row',
		maxHeight: 80,
		marginBottom: 12
	},
	switchTextContainer: {
		flex: 1,
		marginRight: 8
	},
	label: {
		fontSize: 14,
		...sharedStyles.textMedium
	},
	hint: {
		fontSize: 14,
		...sharedStyles.textRegular
	}
});

export interface ISwitch extends SwitchProps {
	id: string;
	label: string;
	hint: string;
	onValueChange: (value: boolean) => void;
}

export const SwitchItem = ({ id, value, label, hint, onValueChange, disabled = false }: ISwitch) => {
	const { colors } = useTheme();

	return (
		<View style={[styles.switchContainer, { backgroundColor: colors.backgroundColor }]}>
			<View style={styles.switchTextContainer}>
				<Text style={[styles.label, { color: colors.titleText }]}>{I18n.t(label)}</Text>
				<Text testID={`create-channel-${id}-hint`} style={[styles.hint, { color: colors.auxiliaryText }]}>
					{I18n.t(hint)}
				</Text>
			</View>
			<Switch
				value={value}
				onValueChange={onValueChange}
				testID={`create-channel-${id}`}
				trackColor={SWITCH_TRACK_COLOR}
				disabled={disabled}
			/>
		</View>
	);
};
