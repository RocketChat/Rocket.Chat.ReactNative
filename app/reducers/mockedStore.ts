import { applyMiddleware, compose, createStore } from 'redux';
import createSagaMiddleware from 'redux-saga';

import reducers from '.';

const enhancers = compose(applyMiddleware(createSagaMiddleware()));
export const mockedStore = createStore(reducers, enhancers);
