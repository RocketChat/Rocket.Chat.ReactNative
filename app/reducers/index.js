import { combineReducers } from 'redux';
import settings from './reducers';
import login from './login';
import meteor from './connect';
import messages from './messages';
import room from './room';
import rooms from './rooms';
import server from './server';
import selectedUsers from './selectedUsers';
import createChannel from './createChannel';
import app from './app';
import customEmojis from './customEmojis';
import activeUsers from './activeUsers';
import roles from './roles';
import starredMessages from './starredMessages';
import pinnedMessages from './pinnedMessages';
import mentionedMessages from './mentionedMessages';
import snippetedMessages from './snippetedMessages';
import roomFiles from './roomFiles';
import sortPreferences from './sortPreferences';

export default combineReducers({
	settings,
	login,
	meteor,
	messages,
	server,
	selectedUsers,
	createChannel,
	app,
	room,
	rooms,
	customEmojis,
	activeUsers,
	roles,
	starredMessages,
	pinnedMessages,
	mentionedMessages,
	snippetedMessages,
	roomFiles,
	sortPreferences
});
