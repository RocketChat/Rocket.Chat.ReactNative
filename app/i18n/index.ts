import i18n from 'i18n-js';
import { I18nManager } from 'react-native';
import * as RNLocalize from 'react-native-localize';
import moment from 'moment';
import 'moment/min/locales';

import { toMomentLocale } from './moment';
import { isRTL } from './isRTL';
import englishJson from './locales/en.json';

type TTranslatedKeys = keyof typeof englishJson;

export { isRTL };

interface ILanguage {
	label: string;
	value: string;
	file: () => any;
}

export const LANGUAGES: ILanguage[] = [
	{
		label: 'English',
		value: 'en',
		file: () => require('./locales/en.json')
	},
	{
		label: 'العربية',
		value: 'ar',
		file: () => require('./locales/ar.json')
	},
	{
		label: 'Deutsch',
		value: 'de',
		file: () => require('./locales/de.json')
	},
	{
		label: 'Español (ES)',
		value: 'es-ES',
		file: () => require('./locales/es-ES.json')
	},
	{
		label: 'Finnish',
		value: 'fi',
		file: () => require('./locales/fi.json')
	},
	{
		label: 'Français',
		value: 'fr',
		file: () => require('./locales/fr.json')
	},
	{
		label: 'Italiano',
		value: 'it',
		file: () => require('./locales/it.json')
	},
	{
		label: '日本語',
		value: 'ja',
		file: () => require('./locales/ja.json')
	},
	{
		label: 'Nederlands',
		value: 'nl',
		file: () => require('./locales/nl.json')
	},
	{
		label: 'Português (BR)',
		value: 'pt-BR',
		file: () => require('./locales/pt-BR.json')
	},
	{
		label: 'Português (PT)',
		value: 'pt-PT',
		file: () => require('./locales/pt-PT.json')
	},
	{
		label: 'Russian',
		value: 'ru',
		file: () => require('./locales/ru.json')
	},
	{
		label: 'Slovenian (Slovenia)',
		value: 'sl-SI',
		file: () => require('./locales/sl-SI.json')
	},
	{
		label: 'Swedish',
		value: 'sv',
		file: () => require('./locales/sv.json')
	},
	{
		label: 'Türkçe',
		value: 'tr',
		file: () => require('./locales/tr.json')
	},
	{
		label: '简体中文',
		value: 'zh-CN',
		file: () => require('./locales/zh-CN.json')
	},
	{
		label: '繁體中文',
		value: 'zh-TW',
		file: () => require('./locales/zh-TW.json')
	}
];

interface ITranslations {
	[language: string]: () => typeof englishJson;
}

const translations = LANGUAGES.reduce((ret, item) => {
	ret[item.value] = item.file;
	return ret;
}, {} as ITranslations);

export const setLanguage = (l: string) => {
	if (!l) {
		return;
	}
	// server uses lowercase pattern (pt-br), but we're forced to use standard pattern (pt-BR)
	let locale = LANGUAGES.find(ll => ll.value.toLowerCase() === l.toLowerCase())?.value;
	if (!locale) {
		locale = 'en';
	}
	// don't go forward if it's the same language and default language (en) was setup already
	if (i18n.locale === locale && i18n.translations?.en) {
		return;
	}
	i18n.locale = locale;
	i18n.translations = { ...i18n.translations, [locale]: translations[locale]?.() };
	I18nManager.forceRTL(isRTL(locale));
	I18nManager.swapLeftAndRightInRTL(isRTL(locale));
	// TODO: Review this logic
	// @ts-ignore
	i18n.isRTL = I18nManager.isRTL;
	moment.locale(toMomentLocale(locale));
};

i18n.translations = { en: translations.en?.() };
const defaultLanguage = { languageTag: 'en', isRTL: false };
const availableLanguages = Object.keys(translations);
const { languageTag } = RNLocalize.findBestAvailableLanguage(availableLanguages) || defaultLanguage;

// @ts-ignore
i18n.isTranslated = (text?: string) => text in englishJson;

setLanguage(languageTag);
i18n.fallbacks = true;

type Ti18n = {
	isRTL: boolean;
	t(scope: TTranslatedKeys, options?: any): string;
	isTranslated: (text?: string) => boolean;
} & typeof i18n;

export default i18n as Ti18n;
