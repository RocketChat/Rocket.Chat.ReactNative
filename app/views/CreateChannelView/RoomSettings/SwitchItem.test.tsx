import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import i18n from '../../../i18n';
import { SwitchItem, ISwitch } from './SwitchItem';
import { mockedStore as store } from '../../../reducers/mockedStore';

const onPressMock = jest.fn((value: boolean) => value);

const testSwitch = {
	id: 'switch-id',
	hint: 'Read_only_hint',
	label: 'Onboarding_title',
	onValueChange: onPressMock,
	value: false,
	testSwitchID: 'create-channel-switch-id',
	testLabelID: 'create-channel-switch-id-hint'
};

const Render = ({ hint, id, label, onValueChange, value }: ISwitch) => (
	<Provider store={store}>
		<SwitchItem hint={hint} id={id} label={label} onValueChange={onValueChange} value={value} />
	</Provider>
);

describe('SwitchItemEncrypted', () => {
	it('should not render the Encrypted Switch component', async () => {
		const { findByTestId } = render(
			<Render
				hint={testSwitch.hint}
				id={testSwitch.id}
				label={testSwitch.label}
				onValueChange={value => testSwitch.onValueChange(value)}
				value={testSwitch.value}
			/>
		);
		const component = await findByTestId(testSwitch.testSwitchID);
		expect(component).toBeTruthy();
	});
	it('should change value of switch', async () => {
		const { findByTestId } = render(
			<Render
				hint={testSwitch.hint}
				id={testSwitch.id}
				label={testSwitch.label}
				onValueChange={value => testSwitch.onValueChange(value)}
				value={testSwitch.value}
			/>
		);
		const component = await findByTestId(testSwitch.testSwitchID);
		fireEvent(component, 'valueChange', { value: true });
		expect(onPressMock).toHaveReturnedWith({ value: !testSwitch.value });
	});
	it('check if hint exists and is the same from testSwitch object', async () => {
		const { findByTestId } = render(
			<Render
				hint={testSwitch.hint}
				id={testSwitch.id}
				label={testSwitch.label}
				onValueChange={value => testSwitch.onValueChange(value)}
				value={testSwitch.value}
			/>
		);
		const component = await findByTestId(testSwitch.testLabelID);
		expect(component.props.children).toBe(i18n.t(testSwitch.hint));
	});
});
