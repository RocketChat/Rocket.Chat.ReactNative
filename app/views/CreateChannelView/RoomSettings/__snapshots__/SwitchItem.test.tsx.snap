// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Snapshots: Switch should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "maxHeight": 80,
          "minHeight": 54,
        },
        {
          "backgroundColor": "#F7F8FA",
        },
      ]
    }
  >
    <View
      accessibilityLabel="[missing "en.Onboarding_title" translation], Only channel owners can send new messages"
      accessible={true}
      style={
        {
          "flex": 1,
          "marginRight": 8,
        }
      }
    >
      <Text
        style={
          [
            {
              "backgroundColor": "transparent",
              "fontFamily": "Inter",
              "fontSize": 16,
              "fontWeight": "500",
              "lineHeight": 24,
              "textAlign": "left",
            },
            {
              "color": "#1F2329",
            },
          ]
        }
      >
        [missing "en.Onboarding_title" translation]
      </Text>
      <Text
        style={
          [
            {
              "backgroundColor": "transparent",
              "fontFamily": "Inter",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "textAlign": "left",
            },
            {
              "color": "#6C727A",
            },
          ]
        }
        testID="create-channel-switch-id-hint"
      >
        Only channel owners can send new messages
      </Text>
    </View>
    <RCTSwitch
      accessibilityRole="switch"
      accessible={true}
      disabled={false}
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#156FF5"
      style={
        [
          {
            "height": 31,
            "width": 51,
          },
          {
            "backgroundColor": "#6C727A",
            "borderRadius": 16,
          },
        ]
      }
      testID="create-channel-switch-id"
      thumbTintColor="#FFFFFF"
      tintColor="#6C727A"
      value={false}
    />
  </View>
</View>
`;
